src/components/play/PlayUI.js

```diff
@@ -1581,21 +1610,711 @@ class PlayUI extends Component {
+            clickableSequenceButton.on('click', this.onToggleRecordSequence)
+            clickableSequenceButton.attr({ id: 'sequence-button', fill: '#000', opacity: 0.00001, cursor: 'pointer' })
+            clickableSequenceButton.x(sStopButtonX)
+            clickableSequenceButton.y(sStopButtonY)
+        }
+    }
+
+    onToggleRecordSequence = () => {
+        const { isPlaying } = this.props
+        if (isPlaying)
+            this.onPlaybackToggle()
+        this.isRecordingSequence = !this.isRecordingSequence
+        this.onRecordSequenceClick()
+        this.renderPatternPresetsSequencer()
+    }
+
+    clearPresetPatternsSequencer = () => {
+
+        const switchLabel = document.getElementById('switch-letter')
+        switchLabel && switchLabel.parentNode.removeChild(switchLabel)
+
+        const switchLabelSubcontainer = document.getElementById('switch-letter-subcontainer')
+        switchLabelSubcontainer && switchLabelSubcontainer.parentNode.removeChild(switchLabelSubcontainer)
+
+        const sequenceSwitch = document.getElementById('sequence-switch')
+        sequenceSwitch && sequenceSwitch.parentNode.removeChild(sequenceSwitch)
+
+        const clickableSwitch = document.getElementById('clickable-switch')
+        clickableSwitch && clickableSwitch.parentNode.removeChild(clickableSwitch)
+
+        const switchLabelContainer = document.getElementById('switch-letter-container')
+        switchLabelContainer && switchLabelContainer.parentNode.removeChild(switchLabelContainer)
+
+        for (let i = 0; i < HTML_UI_Params.sequenceButtonDots; i++) {
+            const currentDot = document.getElementById(`${i}_sequence_dot`)
+            currentDot && currentDot.parentNode.removeChild(currentDot)
+        }
+        this.clearPresetGraphics()
+
+        const sbtn = document.getElementById('sequence-button')
+        sbtn && sbtn.parentNode.removeChild(sbtn)
+
+        const sText = document.getElementById('sequence-text')
+        sText && sText.parentNode.removeChild(sText)
+        for (let i = 0; i < HTML_UI_Params.sequenceButtonDots; i++) {
+            const dot = document.getElementById(`${i}-sbuttonDot`)
+            dot && dot.parentNode.removeChild(dot)
+        }
+        const sStop = document.getElementById('sequence-stop')
+        sStop && sStop.parentNode.removeChild(sStop)
+    }
+
+    clearPresetGraphics = () => {
+        const { round, user } = this.props
+        const patterns = round.userPatterns[user.id].patterns
+        for (let i = 0; i < patterns.length; i++) {
+            const { state: { layers } } = patterns[i]
+            const rnd = document.getElementById(`${i}_pattern`)
+            const lbl = document.getElementById(`${i}_pattern_label`)
+            const btn = document.getElementById(`${i}_pattern_clickable_button`)
+            rnd && rnd.parentNode.removeChild(rnd)
```

Comment:
Sorta goes for all these document. getElementById stuff/remove child

Can't you just update the model? Or a property?

Reply:
I'll find a solution.